imap_processing/spice/pointing_frame.py

```diff
@@ -200,57 +213,72 @@ def calculate_pointing_attitude_segments(
+    # to cover the time range of the new repoint table.
+    # Get the coverage of the CK files storing the earliest start time and
+    # latest end time.
+    et_start = np.inf
+    et_end = -np.inf
+    for ck_path in ck_paths:
+        ck_cover = spiceypy.ckcov(
+            str(ck_path), int(id_imap_spacecraft), True, "INTERVAL", 0, "TDB"
         )
+        num_intervals = spiceypy.wncard(ck_cover)
+        individual_ck_start, _ = spiceypy.wnfetd(ck_cover, 0)
+        _, individual_ck_end = spiceypy.wnfetd(ck_cover, num_intervals - 1)
+        logger.debug(
+            f"{ck_path.name} covers time range: ({et_to_utc(individual_ck_start)}, "
+            f"{et_to_utc(individual_ck_end)}) in {num_intervals} intervals."
+        )
+        et_start = min(et_start, individual_ck_start)
+        et_end = max(et_end, individual_ck_end)
 
-    id_imap_spacecraft = spiceypy.gipool("FRAME_IMAP_SPACECRAFT", 0, 1)
-
-    # Select only the pointings within the attitude coverage.
-    ck_cover = spiceypy.ckcov(
-        str(ck_path), int(id_imap_spacecraft), True, "INTERVAL", 0, "TDB"
-    )
-    num_intervals = spiceypy.wncard(ck_cover)
-    et_start, _ = spiceypy.wnfetd(ck_cover, 0)
-    _, et_end = spiceypy.wnfetd(ck_cover, num_intervals - 1)
     logger.info(
-        f"{ck_path.name} contains {num_intervals} intervals with "
-        f"start time: {et_to_utc(et_start)}, and end time: {et_to_utc(et_end)}"
+        f"CK kernels combined coverage range: "
+        f"{(et_to_utc(et_start), et_to_utc(et_end))}, "
     )
 
-    # Get data from the repoint table and filter to only the pointings fully
-    # covered by this attitude kernel
+    # Get data from the repoint table and convert to Pointings
     repoint_df = get_repoint_data()
     repoint_df["repoint_start_et"] = sct_to_et(
         met_to_sclkticks(repoint_df["repoint_start_met"].values)
     )
     repoint_df["repoint_end_et"] = sct_to_et(
         met_to_sclkticks(repoint_df["repoint_end_met"].values)
     )
-    repoint_df = repoint_df[
-        (repoint_df["repoint_end_et"] >= et_start)
-        & (repoint_df["repoint_start_et"] <= et_end)
-    ]
-    n_pointings = len(repoint_df) - 1
+    pointing_ids = repoint_df["repoint_id"].values[:-1]
+    pointing_start_ets = repoint_df["repoint_end_et"].values[:-1]
+    pointing_end_ets = repoint_df["repoint_start_et"].values[1:]
+
+    # Keep only the pointings that are fully covered by the attitude kernels.
+    keep_mask = (pointing_start_ets >= et_start) & (pointing_end_ets <= et_end)
+    # Filter the pointing data.
+    pointing_ids = pointing_ids[keep_mask]
+    pointing_start_ets = pointing_start_ets[keep_mask]
+    pointing_end_ets = pointing_end_ets[keep_mask]
+
+    n_pointings = len(pointing_ids)
+    if n_pointings == 0:
+        logger.warning(
+            "No Pointings identified based on coverage of this CK file. Skipping."
+        )
+        return np.array([], dtype=POINTING_SEGMENT_DTYPE)
```

Comment:
Looking at this, I think this would all fall out below as well (create an empty array that gets returned and not iterate through the for-loop because it is zero) so it doesn't need a special case here. The warning message seems useful though.